docs/data/data-grid/column-menu/ReuseColumnMenuGrid.js

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import IconFilter from '@mui/icons-material/FilterAlt';
+import {
+  DataGrid,
+  GridColumnMenuDefault,
+  gridColumnMenuSlots,
+} from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+function CustomFilterItem(props) {
+  const { column, onClick } = props;
+  return (
+    <Button sx={{ m: 1 }} onClick={onClick} startIcon={<IconFilter />}>
+      Show {column.field} Filters
+    </Button>
+  );
+}
+
+function CustomColumnMenu(props) {
+  return (
+    <GridColumnMenuDefault
+      {...props}
+      slots={{
+        ...gridColumnMenuSlots,
+        ColumnMenuFilterItem: {
+          // override component for Filter item
+          component: CustomFilterItem,
+          // change order
+          priority: 1,
```

Comment:
Yes, I'll add the priorities for `default` exports and improve the documentation a bit so that users can set the priorities as they want their items ordered.